Jenkins newsletter September 2022 (#5590)
jenkins-infra/jenkins.io: content/blog/2022/10/2022-10-14-jenkins-newsletter.adoc

---
layout: post
title: "Jenkins September 2022 Newsletter"
tags:
- jenkins
- newsletter
- community
- contribute
authors:
- alyssat
opengraph:
  image: /images/post-images/2022-10-14-jenkins-newsletter/image1.png
---

== September 2022

image:/images/post-images/2022-10-14-jenkins-newsletter/image1.png[image,width=289,height=175] Monthly Newsletter

Welcome to the Jenkins Newsletter!
This is a compilation of progress within the project, highlighted by Jenkins Special Interest Groups (SIGs) for the month of September.

Happy reading!

== Security Update

* Two security advisories during September with "special" nature for Jenkins core
** One was only impacting very specifically configured instance
** The other was only impacting three successive weekly releases

== Infrastructure Update 

> * Dropping JDK8 from all agents and controllers processes with the 2.361.1 September LTS

Reviewer: ```suggestion

* Dropping JDK8 from all agents and controllers processes with the 2.361.1 September LTS
```

nitpick: Wether you are using asciidoctor or Markdown, surrounding blocks with an (only 1) empty line is considered a best practise:

- Clarity of the source (when reading/reviewing)
- Help different parsers (IDEs, previews in GitHub, CLIs)

---
layout: post
title: "Jenkins September 2022 Newsletter"
tags:
- jenkins
- newsletter
- community
- contribute
authors:
- atong
opengraph:
  image: /images/post-images/2022-10-14-jenkins-newsletter/image1.png
---

== September 2022

image:/images/post-images/2022-10-14-jenkins-newsletter/image1.png[image,width=289,height=175]

Welcome to the Jenkins Newsletter! This is a compilation of progress
within the project, highlighted by Jenkins Special Interest Groups
(SIGs) for the month of September.

Happy reading!

== Security Update

> * Two security advisories during September with “special” nature for Jenkins core

Reviewer 2: "of special nature"?

** One was only impacting very specifically configured instance
** The other was only impacting three successive weekly releases

== Infrastructure Update 
* Dropping JDK8 from all agents and controllers processes with the 2.361.1
September LTS
* Joint-work with JFrog (our sponsor) to decrease the bandwidth usage (50
Tb per month, should be under 10 Tb per month) of the Jenkins’ Maven
repository repo.jenkins-ci.org
* Digital Ocean renewed their sponsorship with increase credits ($18,000
for 1 year)
* Observability of ci.jenkins.io for plugin developers using Datadog’s
(also a sponsor) metric collection on ephemeral agents. We hope to also
use traces in the future
* New “All-in-one” agent template built with Packer: whether your builds
are running on a VM or a container, in AWS, Azure or Digital Ocean, you
have the same tools in the same locations

== Platform Update

* Loaner computers for ppc64le are gone
** IBM suspended the loaner machine program
* arm/v7 agent Docker images
* Linux packages use systemd instead of System V init
* Exit lifecycle change in the Docker images

> * Container image deprecation for the blueocean container

Reviewer 2: ```suggestion
* Blue Ocean container image deprecation
```

(jenkinsci:blueocean)
* Container repository management for Jenkins agents
** Simplifying the release process
* JDK 11 now required since Jenkins 2.357 and 2.361.1 LTS
** Ongoing minor tasks
*** Dropping Java 8 from the Docker agents - merged, not yet released
*** Plugins beginning to require Java 11, ongoing process
** Jenkins works with Java 17 since 2.355 and 2.346.1 LTS
** Ongoing, a few recent reports, working well
** Jenkins infra planning to put one or more of the infra machines on JDK
17
** Timeline indicates end of Java 11 first level maintenance Sep 2023,
security fixes for several more years after end of first level
maintenance

> ** More than 11 000 Java 17 installs reporting worldwide with Jenkins in

Reviewer 2: ```suggestion
** More than 11,000 Java 17 installs reporting worldwide with Jenkins in
```

September

== Documentation Update

=== DigitalOcean and Jenkins partnership continues to grow!

> Thanks to DigitalOcean’s generosity, Jenkins has access to more resources than ever before, such as utilizing the DigitalOcean platform to continue development and share progress made. We can dedicate our time and resources more fully, allowing users to engage and make an impact. Read the story https://www.jenkins.io/blog/2022/09/19/digital-ocean-sponsorship/[+++HERE+++].

Reviewer 2: ```suggestion
Thanks to DigitalOcean's generosity, Jenkins has access to more resources than ever before, such as utilizing the DigitalOcean platform to continue development and share progress made. We can dedicate our time and resources more fully, allowing users to engage and make an impact. Read the story https://www.jenkins.io/blog/2022/09/19/digital-ocean-sponsorship/[+++on our blog+++].
```

=== image:/images/post-images/2022-10-14-jenkins-newsletter/image6.png[image,width=247,height=139]

===  Java 11 or Newer

Beginning with Jenkins 2.357 (released on June 28, 2022) and the
forthcoming 2.361.1 LTS release, Jenkins requires Java 11 or newer.
Additionally, beginning with Jenkins 2.355 (released on June 14, 2022)
and Jenkins 2.346.1 LTS (released on June 22, 2022), Jenkins supports
Java 17. Plugins have already been prepared in
https://issues.jenkins.io/browse/JENKINS-68446[JENKINS-68446]. Use the
Plugin Manager to upgrade all plugins before _and_ after upgrading to
Jenkins 2.357. Read the post
https://www.jenkins.io/blog/2022/06/28/require-java-11/[+++HERE+++].

=== image:/images/post-images/2022-10-14-jenkins-newsletter/image3.png[image,width=239,height=139]

=== Thinking About Contributing to Open Source?

> Ponder no more! We’ve created the

Reviewer 2: ```suggestion
Ponder no more! We've created the
```

https://www.jenkins.io/doc/developer/tutorial-improve/["Improve a

> plugin"] developer tutorial for new contributors. We’d love to work with

Reviewer 2: ```suggestion
plugin"] developer tutorial for new contributors. We'd love to work with
```

you!

=== About Blue Ocean

> image:/images/post-images/2022-10-14-jenkins-newsletter/image4.png[image,width=624,height=286]

Reviewer 2: I would reproduce the text here instead of a text screenshot.

Author: Done, thanks.

=== SCaLE19X

It was great to have returned to in person events!
Thank you SCaLE for hosting us.
See you again in March 2023!

image:/images/post-images/2022-10-14-jenkins-newsletter/image2.png[image,width=327,height=154]

=== Google Summer of Code Midterm Status Update

2022 GSoC Contributors worked hard to improve the following projects.
In a midterm webinar, they presented their progress, lessons learned and
achievements:

Recording is at: https://youtu.be/loLSNdCv6K4[+++[Jenkins Online
Meetup]: GSoC Midterm Status Update & Demos - July 21, 2022+++]

Slides are at: https://docs.google.com/presentation/d/1t2vuNn1NFpDusnw0m4vdFw6WBQMeU6kccv_K1v2L6R0/edit?usp=sharing[+++Jenkins
Online Meetup - GSoC 2022 Midterm Demos+++]

* Plugin Health Scoring System by
https://github.com/dheerajodha[+++Dheeraj Singh Jodha+++]
* Jenkinsfile Runner Action for GitHub Actions by
https://github.com/Cr1t-GYM[+++Yiming Gong+++]
* Automatic git cache maintenance on the controller by
https://github.com/hrushi20[+++Hrushikesh Rao+++]
* Pipeline Step Documentation Generator Improvements by
https://github.com/vihaanthora[+++Vihaan Thora+++]
* Link to all projects:

> https://www.jenkins.io/projects/gsoc/[+++https://www.jenkins.io/projects/gsoc/+++]

Reviewer 3: ```suggestion
link:/projects/gsoc/[Google Summer of Code in Jenkins]
```

=== Hacktoberfest

*https://github.com/jmMeessen[+++jmMeessen+++]* announced …

September was the perfect time to prepare for Hacktoberfest.
We got a jump start by finding projects to contribute to, adding "Hacktoberfest"
tag to projects, or getting familiarized with Git.

> Get the deets at https://www.jenkins.io/blog/2022/09/15/preptember/[+++https://www.jenkins.io/blog/2022/09/15/preptember/+++]

Reviewer 3: ```suggestion
To get the deets see the link:/blog/2022/09/15/preptember/[blog post about Preptember].
```
